ipatests/test_xmlrpc/test_automember_plugin.py

```diff
@@ -406,6 +508,95 @@ def test_delete_deps_for_rebuilding_hostgroups(self, host1, hostgroup1,
         automember_hostgroup.delete()
 
 
+class TestAutomemberModifyNegativeSameval(XMLRPC_test):
+
+    def test_modify_group_with_same_value(self):
+        "Try to modify existing group rule with same value"
+        group1.check_update()
+        command = automember_group.make_update_command(type=u'group',
+                                                       desc='%s' % group1)
+        with raises_exact(errors.NotFound(
+                reason=u'no modifications to be performed')):
+            command()
+
+    def test_modify_hostgroup_with_same_value(self):
+        "Try to modify existing hostgroup rule with same value"
+        hostgroup1.check_update()
+        command = automember_hostgroup.make_update_command(
+            type=u'hostgroup', desc='%s' % hostgroup1)
+        with raises_exact(errors.NotFound(
+                reason=u'no modifications to be performed')):
+            command()
+
+
+class TestAutomemberModifyNegativeBadgroup(XMLRPC_test):
+
+    def test_modify_with_nonexistent_group(self, automember_group, group1):
+        "Try to modify a rule with non-existent group"
+        group1.check_update()
+        command = automember_group.make_update_command(type=u'group',
+                                                       desc='%s' % group1)
+        with raises_exact(errors.NotFound(
+                reason=u'%s: Automember rule not found' % group1.cn)):
+            command()
+
+    def test_modify_with_nonexistent_hostgroup(self, automember_hostgroup,
+                                               hostgroup1):
+        "Try to modify a rule with non-existent hostgroup"
+        hostgroup1.check_update()
```

Comment:
same as above, make sure it exists or doesn't. It is applicable in multiple locations, please go through the every test and make sure you have the correct error and missing/exists requirement